refactor(scrapers.tasks.update_from_text):  reuse make_citation in update_from_text

grossir opened the pull request:
Solves #4903

- Move make_citation from cl_scrape_opinions into scrapers.utils
- Move citation_is_duplicated from cl_back_scrape_citations into scrapers.utils
- Delete scraped_citation_object_is_valid, now we rely on eyecite, used by make_citation
- Refactor test site to account for changes

flooie commented on cl/scrapers/tasks.py:
```diff
@@ -50,31 +49,32 @@ def update_document_from_text(
-                _, citation_created = ModelClass.objects.get_or_create(**data)
-                metadata_dict["Citation"]["created"] = citation_created
+            citation = make_citation(data, opinion.cluster, court_id)
+            if not citation or citation_is_duplicated(citation, data):
+                continue
```
Would it make more sense to throw the error logging here - and not pass court id into make citation.

grossir replied:
It's awkward to pass the court_id only for logging, but if we didn't do it inside this function we would have to make an outside `logger.error` call everytime `make_citation` returns None

grossir commented:
I fixed the tests; and added a call to `extract_doc_content` to `update_from_text`, for when the text extraction failed. I noticed we weren't covering this use case on the command, and it's actually a very important one.

Please give it another check @flooie , but notice that before we merge this, we would have to:
1. merge the https://github.com/freelawproject/juriscraper/pull/1298 in Juriscraper
2. publish a new juriscraper
3. update the poetry requirement in this PR to that new Juriscraper version

So that it doesn't break anything
I tested it in my local env with the juriscraper branch, and it's working!

```
# copy some PA Super cluster where citation extraction had failed; then

docker exec -it cl-django pip uninstall juriscraper
docker exec -it cl-django pip install https://github.com/freelawproject/juriscraper/archive/extract_from_text_citations.zip

docker exec -it cl-django python /opt/courtlistener/manage.py update_from_text --courts juriscraper.opinions.united_states.state.pasuperct --verbosity 3 --date-filed-gte=2024-12-01 --date-filed-lte=2026-01-01 --cluster-status=Published
```

flooie commented on poetry.lock:
```diff
@@ -5690,6 +5937,6 @@ test = ["coverage[toml]", "zope.event", "zope.testing"]
 
 [metadata]
-lock-version = "2.0"
+lock-version = "2.1"
```
think this is causing the issue?

grossir replied:
It seems there is no issue? 
Tests are passing and they imply building the docker images, so this poetry.lock seems to be working

flooie commented:
@grossir this looks great - and will be great to get these errors gone.
